src/main.ts

```diff
@@ -31,13 +31,11 @@ export async function runMain<T extends ArgsDef = ArgsDef>(
     }
   } catch (error: any) {
     const isCLIError = error instanceof CLIError;
-    if (!isCLIError) {
-      consola.error(error, "\n");
-    }
     if (isCLIError) {
       await showUsage(...(await resolveSubCommand(cmd, rawArgs)));
+    } else {
+      consola.error(error, "\n");
     }
-    consola.error(error.message);
```

Comment:
There are only two conditions, is cli error or not, so we can make sure there must be something printed already, if the error` is undefined the `error.message` should be empty or raise a runtime error. So just remove it to avoid printing duplicated error.